nuxeo-retention/src/main/java/org/nuxeo/retention/service/RetentionManagerImpl.java

```diff
@@ -279,68 +305,100 @@ public void evalRules(Map<String, Set<String>> docsToCheckAndEvents) {
         Framework.getService(WorkManager.class).schedule(work, WorkManager.Scheduling.ENQUEUE);
     }
 
-    protected ELActionContext initActionContext(DocumentModel doc, CoreSession session) {
-        ELActionContext ctx = new ELActionContext(new ExpressionContext(), new ExpressionFactoryImpl());
-        ctx.setCurrentPrincipal(session.getPrincipal());
-        doc.detach(true);
-        ctx.setCurrentDocument(doc);
-        return ctx;
-    }
-
-    protected boolean evaluateConditionExpression(ELActionContext ctx, String expression) {
-        Calendar now = Calendar.getInstance();
+    protected boolean evaluateConditionExpression(Record record, String eventInput, CoreSession session) {
+        String expression = record.getRule(session).getStartingPointExpression();
         if (StringUtils.isEmpty(expression)) {
-            return false;
+            return true;
         }
+        ELActionContext ctx = new ELActionContext(new ExpressionContext(), new ExpressionFactoryImpl());
+        record.getDocument().detach(true);
+        ctx.setCurrentDocument(record.getDocument());
+        Calendar now = Calendar.getInstance();
         ctx.putLocalVariable("currentDate", now);
+        if (StringUtils.isNotBlank(eventInput)) {
+            if (EVENT_INPUT_PATTERN.matcher(eventInput).matches()) {
+                ctx.putLocalVariable(EVENT_BASED_RULE_INPUT, eventInput);
+            } else {
+                log.warn(
+                        "Invalid retention event input: {}, only alphanumeric, dash, underscore and period character are allowed",
+                        eventInput);
+            }
+        }
+        // Only restricted user can fill expression and eventInput
         return ctx.checkCondition(expression);
     }
 
     @Override
-    public void evalExpressionEventBasedRules(Record record, Set<String> events, CoreSession session) {
-        if (record == null) {
-            return; // nothing to do
-        }
+    public boolean evalExpressionEventBasedRules(Record record, String event, String eventInput, CoreSession session) {
         RetentionRule rule = record.getRule(session);
         if (rule == null) {
-            return; // nothing to do
+            return false; // nothing to do
         }
         if (!rule.isEventBased()) {
-            log.trace("Record is not event-based");
-            return;
+            log.trace("Rule {} for Record {} is not event-based", () -> rule.getDocument().getPathAsString(),
+                    () -> record.getDocument().getPathAsString());
+            return false;
         }
-        log.debug("Evaluating event-based rules for record {}", () -> record.getDocument().getPathAsString());
+        log.debug("Evaluating event-based rule {} for record {}", () -> rule.getDocument().getPathAsString(),
```

Comment:
Any reason why "rule" instead of "rules"? There indeed seems to be a single rule, though in the other log messages we use "rules".

Reply:
fixed